src/ui/components/common/content_actions/report_button.rs

```diff
@@ -24,27 +21,16 @@ fn report_content(
 }
 
 #[component]
-pub fn ReportButton(
-  creator_name: StoredValue<String>,
-  post_or_comment_id: PostOrCommentId,
-  creator_actor_id: StoredValue<String>,
-) -> impl IntoView {
+pub fn ReportButton<'a>(creator: &'a Person, post_or_comment_id: PostOrCommentId) -> impl IntoView {
   let report_content_label = if matches!(post_or_comment_id, PostOrCommentId::Comment(_)) {
     "Report comment"
   } else {
     "Report post"
   };
+  let onclick = report_content(creator, post_or_comment_id);
```

Comment:
I was running into a bug when I tested your changes locally, and I believe the change to how the `onclick` function is handled is the cause. When I log in as a user, the report modal immediately pops up on the first post in the list. This doesn't happen in main. That's because this change makes it so that the `report_content` function is called when the component is rendered as opposed to when the button is clicked. If you want to move the `onclick` closure to a variable instead of inlining it when passed to the button, this sould be:

```rust
let onclick = move |_| report_content(creator, post_or_comment_id);
```

Then you can just pass it directly to the button:

```html
<button class="text-xs whitespace-nowrap" type="button" on:click=onclick>
```

Reply:
Nice, done.